static/libs-widget-ext.ts

```diff
@@ -0,0 +1,557 @@
+
+        const result = $($(template.children()[0].cloneNode(true)));
+        result.find('.lib-name').html(lib.name);
+        if (!lib.description) {
+            result.find('.lib-description').hide();
+        } else {
+            result.find('.lib-description').html(lib.description);
+        }
+        result.find('.lib-website-link').attr('href', lib.url ?? '#');
+
+        this.conjureUpExamples(result, lib);
+
+        const faveButton = result.find('.lib-fav-button');
+        const faveStar = faveButton.find('.lib-fav-btn-icon');
+        faveButton.hide();
+
+        const versions = result.find('.lib-version-select');
+        versions.html('');
+        const noVersionSelectedOption = $('<option value="">-</option>');
+        versions.append(noVersionSelectedOption);
+        let hasVisibleVersions = false;
+
+        for (let versionId in lib.versions) {
+            const version = lib.versions[versionId];
+            const option = $('<option>');
+            if (version.used) {
+                option.attr('selected', 'selected');
+
+                if (this.isAFavorite(libId, versionId)) {
+                    faveStar.removeClass('far').addClass('fas');
+                }
+
+                faveButton.show();
+            }
+            option.attr('value', versionId);
+            option.html(version.version);
+            if (version.used || !version.hidden) {
+                hasVisibleVersions = true;
+                versions.append(option);
+            }
+        }
+
+        if (!hasVisibleVersions) {
+            noVersionSelectedOption.text('No available versions');
+            versions.prop('disabled', true);
+        }
+
+        faveButton.on('click', () => {
+            const option = versions.find('option:selected');
+            const verId = option.attr('value');
+            if (this.isAFavorite(libId, verId)) {
+                this.removeFromFavorites(libId, verId);
+                faveStar.removeClass('fas').addClass('far');
+            } else {
+                this.addToFavorites(libId, verId);
+                faveStar.removeClass('far').addClass('fas');
+            }
+            this.showFavorites();
+        });
+
+        versions.on('change', () => {
+            const option = versions.find('option:selected');
+            const verId = option.attr('value');
+
+            this.selectLibAndVersion(libId, verId);
+            this.showSelectedLibs();
+
+            if (this.isAFavorite(libId, verId)) {
+                faveStar.removeClass('far').addClass('fas');
+            } else {
+                faveStar.removeClass('fas').addClass('far');
+            }
+
+            if (verId) {
+                faveButton.show();
+            } else {
+                faveButton.hide();
+            }
+
+            this.onChange();
+        });
+
+        return result;
+    }
+
+    addSearchResult(libId: string, library: Library) {
+        // Explain
```

Comment:
I know the difference between `JQuery<Node>` and `JQuery<HTMLElement>`, but without the any here so that the compiler ignores the types, it complains about `JQuery.Node being different from `Node` which, ok.
But then, what's the proper solution here?